Update Framework interfaces and mark as API

mrtuvn opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

Mark following interfaces as API  + removed deprecated methods usage in code base + added declare strict_types
Update integration tests coverage with changes

lib/internal/Magento/Framework/App/Config/ValueInterface.php @api 
lib/internal/Magento/Framework/MessageQueue/ConsumerConfigurationInterface.php @api + remove deprecated method
lib/internal/Magento/Framework/View/DesignInterface.php mark @api + remove deprecated method setArea from interface 
lib/internal/Magento/Framework/Webapi/CustomAttributeTypeLocatorInterface.php @api + remove deprecated method

### Related Pull Requests
<!-- related pull request placeholder -->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/magento2#32037

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->
CC: @ihor-sviziev @sivaschenko @kandy @pauliarush
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

bubasuma committed:
MC-34522: Two near-simultaneous product imports interfere with each other

- Add user id to "import data" table in order to allow concurrent import

bubasuma committed:
MC-34522: Two near-simultaneous product imports interfere with each other

- Change default value of user_id column to NULL for backward compatibility

bubasuma committed:
MC-34522: Two near-simultaneous product imports interfere with each other

- Check table name before adding user_id constraint

bubasuma committed:
MC-34522: Two near-simultaneous product imports interfere with each other

- Fix code style

mrtuvn commented on app/code/Magento/Theme/Model/View/Design.php:
```diff
@@ -105,6 +105,7 @@ public function __construct(
+     * @deprecated
      */
     public function setArea($area)
     {
```
@sivaschenko @sidolov @gabrieldagama  should we safe for remove this method ? Method defined in interface already removed but i just think we should keep method here for backward compatible

Den4ik replied:
In my opinion we should clear legacy code in major release

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

mrtuvn commented on app/code/Magento/Eav/Model/TypeLocator/SimpleType.php:
```diff
@@ -62,14 +62,4 @@ public function getType($attributeCode, $entityType)
-    {
-        $this->serviceTypeList->getDataTypes();
-    }
 }
```
remove public methods in interfaces may violates rule BC principles
https://devdocs.magento.com/contributor-guide/backward-compatible-development/

sivaschenko replied:
@mrtuvn that's why we are targeting these changes to 2.5-develop, that's the only way to remove deprecated public methods

mrtuvn replied:
It's safe if i remove all usage of deprecated methods. Ex: setArea i found used in some place

mrtuvn replied:
removed deprecated mention method

ihor-sviziev commented:
@magento run all tests

mrtuvn commented:
@magento run Integration Tests, Static Tests, Unit Tests, Functional Tests B2B, Functional Tests CE, Functional Tests EE

ihor-sviziev commented on app/code/Magento/Theme/Model/View/Design.php:
```diff
@@ -100,19 +100,6 @@ public function __construct(
-     */
-    public function setArea($area)
-    {
-        $this->_area = $area;
```
It seems like this property shouldn't be used anymore and should be removed. What do you think?

mrtuvn replied:
I'm not sure remove that property can make something else break

ihor-sviziev replied:
ok. let's leave it as is

mrtuvn replied:
oh no i have removed that @@ Let's see some results tests

ihor-sviziev reviewed (changes requested):
Now we have following error during `bin/magento setup:static-content:deploy -f en_US`:
Error happened during deploy process: Area code is not set


Here is result: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/32142/47a096751b99127bb2bca87090ccd4a5/Functional/console-error-logs.html

mrtuvn commented:
@sivaschenko any reason why we need remove deprecated method 'setArea' in interface `Magento\Framework\View\DesignInterface`

sivaschenko commented:
^ @paliarush can you please take a look at this

ihor-sviziev commented:
@magento run all tests

mrtuvn commented:
@magento run all tests

ihor-sviziev commented:
Waiting for answer https://github.com/magento/magento2/pull/32142#issuecomment-780497734 https://github.com/magento/magento2/pull/32142#issuecomment-780662414
For now moving to on hold

mrtuvn committed:
Update Framework interfaces and mark as API


update


update fix unit tests


update


update


update


remove property 


up


update


update

mrtuvn commented:
@magento run all tests

mrtuvn commented:
@magento run Functional Tests EE

mrtuvn commented:
How can i pass Sematic check after removed public methods in interfaces as purpose of this PR ?

mrtuvn commented:
@magento run Functional Tests B2B

ihor-sviziev commented:
@mrtuvn I believe that's correct behavior, we'll need to get approval for that

ihor-sviziev commented on app/code/Magento/Theme/Model/View/Design.php:
```diff
@@ -138,7 +138,7 @@ public function setDesignTheme($theme, $area = null)
         if ($area) {
             $this->setArea($area);
-        } else {
+        } elseif (null == $area) {
```
could you explain why it's needed? Also not clear how it worked before

mrtuvn replied:
I think we should avoid use `else` in this case -> so go with elseif for check in case `$area` equal null. In case method `setDesignTheme` not pass second argument. Null default value will always be used. So magento will need to call `getArea` instead setArea

Den4ik replied:
@mrtuvn What about situation when custom module pass empty string as `$area` for example?

magento-engcom-team commented:
Hi @ihor-sviziev, thank you for the review.
[ENGCOM-8837](https://jira.corp.magento.com/browse/ENGCOM-8837) has been created to process this Pull Request
:eight_spoked_asterisk: @ihor-sviziev, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

mrtuvn commented:
@magento run Semantic Version Checker, Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
Failed B2B related external repo inventory

MC-40489: Billing address fields should be cleared after refreshing checkout page with store pickup delivery method
"Field by name, label, CSS or XPath element with .payment-method._active .billing-address-form input[name='firstname'] was not found."
File related
app/code/Magento/InventoryInStorePickupFrontend/Test/Mftf/Test/StorefrontBillingAddressClearFieldsAfterRefreshingPageTest.xml

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
CC: @engcom-Oscar @kate-kyzyma some test case seem not work after this change

mrtuvn commented:
This one still same branch 2.5 target because update changes

mrtuvn commented:
@magento run Functional Tests B2B

mrtuvn commented:
@magento run Semantic Version Checker

Den4ik reviewed (changes requested):
@mrtuvn Thanks for work on this PR.
Could you please move all changes for `dev/tests/integration/testsuite/Magento/Theme/Model/View/DesignTest.php` from https://github.com/magento/magento2/pull/32921 into this PR, because as I see you are clearing same files

mrtuvn committed:
Update code and remove setArea deprecated method

mrtuvn commented:
@magento run all tests

Den4ik commented:
@magento run Sample Data Tests EE, Sample Data Tests B2B, Sample Data Tests CE, Integration Tests, Functional Tests EE, Functional Tests B2B, Functional Tests CE

Den4ik commented:
@magento run Sample Data Tests EE, Sample Data Tests B2B, Sample Data Tests CE, Integration Tests, Functional Tests EE, Functional Tests B2B, Functional Tests CE

Den4ik commented:
I'm not sure, but something strange on builds of 2.5-develop branch. I see same fails on multiple PR to 2.5. @sivaschenko Could you check please?

mrtuvn commented:
Maybe this branch out of sync with 2.4. Supprised this changes may break some internal

mrtuvn commented:
@magento run Integration Tests, Functional Tests EE, Functional Tests B2B, Functional Tests CE

mrtuvn commented:
@magento run all tests

Den4ik commented:
@gabrieldagama do we have information about tests for 2.5-develop branch? There are many failed tests

magento-engcom-team commented:
Hi @Den4ik, thank you for the review.
[ENGCOM-8837](https://jira.corp.magento.com/browse/ENGCOM-8837) has been created to process this Pull Request

mrtuvn commented on app/code/Magento/Theme/Model/Theme.php:
```diff
@@ -273,7 +275,18 @@ public function getArea()
     }
```
can we refactor this method to simply ? Just call directly return getData('area'), Remove redundance code above

CC: @ihor-sviziev

tuna2smc replied:
Just curious should use Magento\Framework\App\Area::PARAM_AREA instead string `area`

mrtuvn replied:
Just curious should use Magento\Framework\App\Area::PARAM_AREA instead string area

mrtuvn committed:
Rework method getData area

mrtuvn commented:
@magento run all tests

mrtuvn committed:
Update Area.php

mrtuvn committed:
Update Theme.php

mrtuvn committed:
Update visibility const

mrtuvn committed:
Merge branch '2.4-develop' into api-interfaces-mark-issue-32037

mrtuvn committed:
Merge branch '2.4-develop' into api-interfaces-mark-issue-32037

mrtuvn commented:
@magento run all tests

mrtuvn committed:
Merge branch '2.4-develop' into api-interfaces-mark-issue-32037

mrtuvn commented:
@magento run all tests

mrtuvn committed:
Update code

mrtuvn committed:
Merge branch '2.4-develop' into api-interfaces-mark-issue-32037

mrtuvn commented:
@magento run all tests
